src/component/archive/ArchiveTable/ArchiveTable.tsx

```diff
@@ -0,0 +1,329 @@
+        setGlobalFilter,
+        setHiddenColumns,
+    } = useTable(
+        {
+            columns: columns as any,
+            data: data as any,
+            initialState,
+            sortTypes,
+            autoResetGlobalFilter: false,
+            autoResetSortBy: false,
+            disableSortRemove: true,
+            disableMultiSort: true,
+        },
+        useGlobalFilter,
+        useSortBy,
+        useFlexLayout
+    );
+
+    useEffect(() => {
+        const hiddenColumns = ['description'];
+        if (isMediumScreen) {
+            hiddenColumns.push('lastSeenAt', 'status');
+        }
+        if (isSmallScreen) {
+            hiddenColumns.push('type', 'createdAt');
+        }
+        setHiddenColumns(hiddenColumns);
+    }, [setHiddenColumns, isSmallScreen, isMediumScreen]);
+
+    useEffect(() => {
+        const tableState: PageQueryType = {};
+        tableState.sort = sortBy[0].id;
+        if (sortBy[0].desc) {
+            tableState.order = 'desc';
+        }
+        if (globalFilter) {
+            tableState.search = globalFilter;
+        }
+
+        setSearchParams(tableState, {
+            replace: true,
+        });
+        setStoredParams({ id: sortBy[0].id, desc: sortBy[0].desc || false });
+    }, [sortBy, globalFilter, setSearchParams, setStoredParams]);
+
+    const [firstRenderedIndex, lastRenderedIndex] =
+        useVirtualizedRange(rowHeight);
+
+    const renderRows = () => {
+        return (
+            <>
+                {rows.map((row, index) => {
+                    const isVirtual =
+                        index < firstRenderedIndex || index > lastRenderedIndex;
+
+                    if (isVirtual) {
+                        return null;
+                    }
+
+                    prepareRow(row);
+                    return (
+                        <TableRow hover {...row.getRowProps()}>
+                            {row.cells.map(cell => (
+                                <TableCell
+                                    {...cell.getCellProps({
+                                        style: {
+                                            flex: cell.column.minWidth
+                                                ? '1 0 auto'
+                                                : undefined,
+                                        },
+                                    })}
+                                    className={classes.cell}
+                                >
+                                    {cell.render('Cell')}
+                                </TableCell>
+                            ))}
+                        </TableRow>
+                    );
+                })}
+            </>
+        );
+    };
+
+    return (
+        <PageContent
+            isLoading={loading}
+            header={
+                <PageHeader
+                    title={`${
+                        inProject ? 'Project Features Archive' : 'Archived'
```

Comment:
Since we have a container for each of these, couldn't the title also be passed in from the outside instead of the `inProject` prop? Then the ArchiveTable truly would keep no internal state knowledge about where it is used.